backend/djangoindia/db/admin.py

```diff
@@ -580,21 +582,58 @@ def send_email_view(self, request):
                     from_email = settings.DEFAULT_FROM_EMAIL
 
                     registration_ids = request.GET.get("ids").split(",")
+                    if not registration_ids:
+                        messages.error(request, "No registration IDs provided.")
+                        return redirect("../")
+
                     queryset = EventUserRegistration.objects.filter(
                         id__in=registration_ids
                     )
+                    if not queryset.exists():
+                        messages.error(request, "No valid registrations found.")
+                        return redirect("../")
+
+                    # List for storing communication
+                    communications = []
 
+                    # Create EventCommunication entries and prepare emails
                     for registration in queryset:
                         recipient_email = registration.user.email
+                        event = registration.event
+
+                        # Create EventCommunication entry
+                        communication = EventCommunication(
+                            event=event,
+                            recipient=registration.user,
+                            subject=subject,
+                            body=message,
+                            status=EventCommunication.STATUS.PENDING,
+                        )
+                        communication.save()
+                        communications.append(communication)
```

Comment:
store recipient ids and set all of them at once, since it will become an M2M field